### PR TITLE
uast: more TypeDeclaration roles

### PR DESCRIPTION
**Pull request by juanjux**
Some additional common roles to map TypeDeclaration children.

### CODE REVIEW

File: uast/uast.go [hunk 1]
```diff
@@ -187,8 +187,15 @@ const (
 	FunctionDeclarationVarArgsList
 
 	// TypeDeclaration is the declaration of a type. It could be a class or
-	// interface in Java, a struct, interface or alias in Go, etc.
+	// interface in Java, a struct, interface or alias in Go, etc. Except for pure or forward declarations
+	// it will usually have a TypeDeclarationBody child and for OOP languages a TypeDeclarationBases and/or
+	// TypeDeclarationInterfaces
```
Comment (alcortesm) [inline uast/uast.go]: Add full stop at the end.

Comment (juanjux) [inline uast/uast.go]: Fixed ::facepalm::

**Comment by smola** [comment]
@juanjux Please, rebase and squash into a single commit.

**Comment by juanjux** [comment]
@smola done, will also do for the other PRs

**Pull request merged**